packages/hardhat-zksync-deploy/src/deployer.ts

```diff
@@ -147,13 +147,24 @@ export class Deployer {
      * @returns Factory dependencies in the format expected by SDK.
      */
     async extractFactoryDeps(artifact: ZkSyncArtifact): Promise<string[]> {
+        const visited = new Set<string>();
+        visited.add(`${artifact.sourceName}:${artifact.contractName}`);
+        return await this.extractFactoryDepsRecursive(artifact, visited);
+    }
+
+    private async extractFactoryDepsRecursive(artifact: ZkSyncArtifact, visited: Set<string>): Promise<string[]> {
```

Comment:
Is it possible to write a test for that?

Reply:
I modified the factory contracts in solidity examples folder, which is tested for compilation and deployment already

Reply:
Yeah, sure, I meant the unit test for the particular method (it seems to be appropriate here).
Not going to insist though.